src/com/pensio/api/request/ReserveSubscriptionChargeRequest.java

```diff
@@ -2,12 +2,29 @@
 
 import com.pensio.Amount;
 
-public class ReserveSubscriptionChargeRequest
+public class ReserveSubscriptionChargeRequest<T extends ReserveSubscriptionChargeRequest<T>>
 {
 	private String agreementId;
 	private AgreementUnscheduledType  agreementUnscheduledType;
 	private Amount amount;
-	
+    private PaymentInfos paymentInfos;
+
+    {
+        paymentInfos = new PaymentInfos();
+    }
+
+    public PaymentInfos getPaymentInfos()
+    {
+        return paymentInfos;
+    }
+
+    @SuppressWarnings("unchecked")
+    public T addPaymentInfo(String key, String value)
```

Comment:
Why are returning T, if the object is strictly String ?

Reply:
T is used here to support type-safe chaining in derived classes, consistent with similar classes using paymentInfo. This allows addPaymentInfo to return the subclass type rather than just ReserveSubscriptionChargeRequest.